modules/cloudfront.js

```diff
@@ -0,0 +1,221 @@
+        allowAll: 'allow-all'
+    },
+    httpMethods: {
+        getOnly: ['HEAD', 'GET'],
+        getAndPost: ['HEAD', 'DELETE', 'POST', 'GET', 'OPTIONS', 'PUT', 'PATCH']
+    },
+    TTLs: {
+        min: 0,
+        max: 31536000,
+        default: 86400
+    },
+    newSite: {
+        headersToKeep: ['Accept', 'Host'],
+        cookies: {
+            Forward: 'whitelist',
+            WhitelistedNames: {
+                Items: cookiesInUse,
+                Quantity: cookiesInUse.length
+            }
+        }
+    },
+    legacy: {
+        headersToKeep: ['*'],
+        cookies: {
+            Forward: 'all'
+        }
+    }
+};
+
+// create a JSON object configured for the legacy/new paths
+const makeBehaviourItem = (origin, path, isPostable, allowQueryStrings, originServer) => {
+    // the new site is properly cached, the legacy is not
+    // so anything legacy should not cache cookies, headers, etc
+    const isLegacy = origin !== 'newSite';
+    const cacheConfig = isLegacy ? BehaviourConfig['legacy'] : BehaviourConfig['newSite'];
+
+    // strip trailing slashes
+    // fixes /welsh => /welsh/ homepage confusion
+    // but doesn't break root/homepage '/' path
+    path = utilities.stripTrailingSlashes(path);
+
+    // use all HTTP methods for legacy
+    const allowedHttpMethods =
+        isLegacy || isPostable ? BehaviourConfig.httpMethods.getAndPost : BehaviourConfig.httpMethods.getOnly;
+    // allow any protocol for legacy, redirect to HTTPS for new
+    const protocol = isLegacy ? BehaviourConfig.protocols.allowAll : BehaviourConfig.protocols.redirectToHttps;
+
+    return {
+        TrustedSigners: {
+            Enabled: false,
+            Items: [],
+            Quantity: 0
+        },
+        LambdaFunctionAssociations: {
+            Items: [],
+            Quantity: 0
+        },
+        TargetOriginId: originServer,
+        ViewerProtocolPolicy: protocol,
+        ForwardedValues: {
+            Headers: {
+                Items: cacheConfig.headersToKeep,
+                Quantity: cacheConfig.headersToKeep.length
+            },
+            Cookies: cacheConfig.cookies,
+            QueryStringCacheKeys: {
+                Items: [],
+                Quantity: 0
+            },
+            QueryString: isLegacy || allowQueryStrings
+        },
+        MaxTTL: BehaviourConfig.TTLs.max,
+        PathPattern: path,
+        SmoothStreaming: false,
+        DefaultTTL: BehaviourConfig.TTLs.default,
+        AllowedMethods: {
+            Items: allowedHttpMethods,
+            CachedMethods: {
+                Items: ['HEAD', 'GET'],
+                Quantity: 2
+            },
+            Quantity: allowedHttpMethods.length
+        },
+        MinTTL: BehaviourConfig.TTLs.min,
+        Compress: false
+    };
+};
+
+const generateUrlList = () => {
+    let urlList = _.cloneDeep(URLs);
+
+    // add auto URLs from route config
+    for (let s in routes.sections) {
```

Comment:
Wonder if it makes sense to pull this out into a form we could write a unit test for, given it's complexity 🤔